File: app/src/main/java/dev/ragnarok/fenrir/api/model/VKApiAudioArtist.java
package dev.ragnarok.fenrir.api.model;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class VKApiAudioArtist implements VKApiAttachment {
    @Nullable
    @SerializedName("id")
    public String id;

    @Nullable
    @SerializedName("name")
    public String name;

    @Nullable
    @SerializedName("photo")
    public List<Image> photo;

    public VKApiAudioArtist() {

    }

    @NonNull
    @Override
    public String getType() {
        return TYPE_ARTIST;
    }

    public static final class Image {
        @Nullable
        @SerializedName("url")
        public String url;

        @SerializedName("width")
        public int width;

        @SerializedName("height")
        public int height;
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/model/VKApiAudioArtist.kt
package dev.ragnarok.fenrir.api.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class VKApiAudioArtist : VKApiAttachment {
    @SerialName("id")
    var id: String? = null

    @SerialName("name")
    var name: String? = null

    @SerialName("photo")
    var photo: List<Image>? = null
    override fun getType(): String {
        return VKApiAttachment.TYPE_ARTIST
    }

    @Serializable
    class Image {
        @SerialName("url")
        var url: String? = null

        @SerialName("width")
        var width = 0

        @SerialName("height")
        var height = 0
    }
}